script/zip-comp-logs

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+"""
+A script to zip the logs produced by a series of competition matches.
+"""
+
+import re
+import shutil
+import argparse
+import datetime
+import tempfile
+from pathlib import Path
+
+
+def zip_folder(folder: Path, args: argparse.Namespace) -> None:
+    tla = folder.name
+    zip_name = f'team-{tla}'
+
+    if args.suffix:
+        zip_name += f'-{args.suffix}'
+    if not args.no_date:
+        zip_name += f'-{datetime.date.today()}'
+
+    shutil.make_archive(
+        f'{args.output_dir / zip_name}',
+        'zip',
+        root_dir=folder,
+        base_dir='.',
+    )
+
+
+def is_team_logs_folder(folder: Path) -> bool:
+    # Options for detecting which folders are logs:
+    # 1. everthing except recordings, matches
+    # 2. folders w/ matching zip file names
+    # 3. regex of folder name
+    # 4. regex of folder contents names
+
+    # Using option 3
+    # folder names are three uppercase letters and an optional number
+    if re.match(r'[A-Z]{3}[0-9]*$', folder.name):
+        return True
+
+    return False
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory containing folders of the teams' robot logs, "
+            "named for the teams' TLAs. This directory also contains the "
+            "teams' robot code, as Zip archives and the match recordings."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        'output_dir',
+        help=(
+            "The directory to place the resulting log Zip archives. "
+            "This should be an empty directory."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        '--suffix',
+        help=(
+            "Adds a suffix to the Zip archive name. "
+            "Archive names follow the format: team-<TLA>-<suffix>-<date>.zip"
+        ),
+        default="",
+    )
+    parser.add_argument(
+        '--no-date',
+        help=(
+            "Remove the date from the archive name."
+        ),
+        action='store_true',
+    )
```

Comment:
I suggest making this (at least internally) be expressed in the positive. I forget the exact syntax, but you can do something like:
```suggestion
    parser.add_argument(
        '--no-date',
        dest='include_date',
        help=(
            "Remove the date from the archive name."
        ),
        action='store_false',
        default=True,
    )
```
Such that usage in code is like `args.include_date` (which therefore avoids double-negatives), but the usage in the CLI is still the way round which is probably more obvious.

Reply:
I was think this but had forgotten about the dest argument. Solved in 6fe7b2b9062397dbee75803b3c7406aafdb7c836